src/ipc/ipc4/handler.c

```diff
@@ -536,6 +536,9 @@ static void ipc_compound_msg_done(uint32_t msg_id, int error)
 
 static int ipc_wait_for_compound_msg(void)
 {
+#if  CONFIG_LIBRARY
```

Comment:
Can we have an inline comment explaining this, I would also make this a separate function for library via the ifdef.